### ISSUE TITLE
Feat: Reconsider the font used in the whole interface

### ISSUE THREAD

**Issue by raphj** [tracim/tracim#3479]
## Feature description and goals

As a first version, we will change the interface font to Nunito, which is a websafe and smoother font than the current one. (PR #5978)
After this first improvement, we will continue the modifications, improving the hierarchy of the texts and maybe adding a new font for specific places.
At the same time, a documentation will be created for our community space teaching people how to block the font to use the browser's base font.

-----------------------------------------------------

# Original issue

## Feature description and goals

To do:
 - [ ] make the used font configurable the same way as the primary color
 
## Context

Tracim currently uses Quicksand in the whole interface. This font arguably has some potential issues prompting us to reconsider this choice and decide whether we should switch to another font, mainly:

- The font is thin, harder to read than the default font in a lot of browsers.
- The font also seems smaller than a regular sans serif font
- The font does not seem optimized for screen with a normal DPI: the shapes do not seem to align with the grid of pixels of these screens, making many letters a bit blur like the `l` (lowecase `L`), `i`, `u`, `m`, `d` for instance. It can also feel "grey" on normal DPI screens (lower contrast) because of this.
- (it is a 100 k dependency that we impose to the user, delaying Tracim's first load)
- Using a font that is different from the one used everywhere else on the user's computer or browser:
   - can make Tracim stand out, which can be both good or bad. A user cannot have particularly bad feeling about an app using the default font used everywhere else, but can dislike an original / weird font (or like it of course, it goes both ways)
   - can make Tracim feel less integrated (arguably bad).

Here's a screenshot of Tracim in its current version on a regular screen, using Quicksand:
![image](https://user-images.githubusercontent.com/3817365/91735683-ed846300-ebac-11ea-850a-9020334f278d.png)

Here's a screenshot of Tracim using `sans-serif` on the same screen, on openSUSE Tumbleweed, for comparison:
![image](https://user-images.githubusercontent.com/3817365/91735539-bdd55b00-ebac-11ea-9fc9-d41b7d5d2cde.png)

Past considerations that are not relevant anymore:
 - #2763 (development files fetched the font from the Google servers)
 - #3468 (bold did not render well on Firefox)

Related Issues

- #3478

## To be discussed

 - [ ] whether we should get rid of Quicksand at all. If we don't, we could still consider using another font weight.
 - [ ] if we do, decide which font we should use?
    - we could try to use the default sans-serif font of the user's browser.
    - we could force a some sans-serif font that looks ok everywhere (Liberation Sans?)
 - [ ] what to test, how should we ensure that changing the font does not break Tracim
 - [ ] find out on what grounds this fond has been chosen and try to respect the initial requirements that are still relevant  
 - [ ] whether we should conduct a user study. But see https://www.nngroup.com/articles/first-rule-of-usability-dont-listen-to-users/ and "Don't make me think - Steve Krug".
 - [ ] Tracim displays documents. On UIs, it's often good practice to use a unique font. On documents, it is usual to use a different font for titles and contents to further separate the titles from the content. For instance, a serif font for titles and a sans-serif font for the content. On regular screen, a serif font would probably lack readability for the content.

Using the default sans-serif font has some advantages:
  - Using a font the user is used to could make the UI feel more familiar, less surprising.
  - The user could have tweaked their default font. Using that would help us respect their choice.
  - The default font is probably designed to look good and to be optimized for readability on the device it is displayed on. 
This also has drawbacks:
  - default fonts can vary, making Tracim more prone to responsive bugs. On the other hand, the same font is already rendered differently across browsers and systems
  - Tracim would not look the same across platforms. However It already looks differently on different screen sizes.

## Extra information

I (@raphj) have been using Tracim for several months while blocking external fonts so I didn't initially notice it was using Quicksand. Tracim already looks good and not broken to me without Quicksand (icons were broken because of Font Awesome being blocked as well which led me to eventually allow external fonts on Tracim).

# See also

#3963 - should be done at the same time

**Comment by raphj** [comment]
here are some font choices of some tools (in this order of fallback):

Github: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji
Mattermost: Open Sans, sans-serif
Nextcloud: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Cantarell, Ubuntu, "Helvetica Neue", Arial, "Noto Color Emoji", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"

Notes:
-  `-apple-system` is the default font on Apple systems
- `Segoe UI` is the default font on Windows for UIs
-  Roboto is the default font on Android and is pretty much like Noto Sans
- Helvetica is the former default font on Mac (isn't anymore)
- Oxygen-Sans is the default font on KDE
- Cantarell is the default font on Gnome 
- sans-serif is the default font of the browser and is mandatory
- Noto Color Emoji is the default emoji font on Android and GNU/Linux
- "Apple Color Emoji" is the default emoji font on Mac systems
- "Segoe UI Emoji" is the default emoji font on Windows

Putting emoji fonts after the sans-serif font makes sense because it overrides some glyphs. But it also makes sense to put it *before* because of this exact reason: some unicode characters like "☺" have been repurposed as Emoji. Problem is, they should be displayed as emojis, but since they are "old", they are in default non-emoji systems fonts.

**Comment by gbonaspetti** [comment]
Hi @raphj , I hope you are well. 
We are working on the subject of this issue and I would like to know if there are any problems you faced (and are not already in this issue) using another font in Tracim. 
If it is not too much trouble, I would also like to know the method you used to block the font.

**Comment by raphj** [comment]
Hello @gbonaspetti I'm good :-) Hope you too, and the whole team. I would not say no to a beer with you all some time.

I didn't spot any issue whatsoever blocking the font and using the system one. Here are the two methods I've used to block fonts:

1. using ublock origin, there is a feature to block fonts. I have it enabled by default but it can be set per website. Unfortunately, it also blocks font awesome icons.
![image](https://user-images.githubusercontent.com/3817365/197349511-344becf8-33c8-463d-b726-91491325b18a.png)

2. using ublock origin, by setting a filter that blocks the address / sets the font
![image](https://user-images.githubusercontent.com/3817365/197349638-128d6db4-7fb8-40c4-98cc-acb0ab0d541a.png)

I happen to still have the two rules, so I can share it:

```
*##.tracim:style(font-family: sans-serif)
/*Quicksand*
```

Hope that helps.

**Comment by gbonaspetti** [comment]
Same here. Cool! Thanks a lot.

**Comment by Millefeuille42** [comment]
Tested on branch
